### ISSUE TITLE
Several python distribtests broken on Linux (on Kokoro)

### ISSUE THREAD

**Issue by jtattermusch** [grpc/grpc#14762]
https://sponge.corp.google.com/target?id=9251c3a1-83ac-4e7e-ab80-0c3a5a0ede10&target=github/grpc&searchFor=&show=ALL&sortBy=STATUS

Most of them fail with 
```
    Could not find <Python.h>. This could mean the following:
      * You're on Ubuntu and haven't run `apt-get install python-dev`.
      * You're on RHEL/Fedora and haven't run `yum install python-devel` or
        `dnf install python-devel` (make sure you also have redhat-rpm-config
        installed)
      * You're on Mac OS X and the usual Python framework was somehow corrupted
        (check your environment variables or try re-installing?)
      * You're on Windows and your Python installation was somehow corrupted
        (check your environment variables or try re-installing?)
```
which AFAIK is basically an indication that the grpcio installation has fallen back to installing grpcio by compiling it from source - which should not happen and a binary wheel should be used.

I assume the tests are equally broken on Jenkins.

**Comment by jtattermusch** [comment]
@kpayson64 can we fix this soon? Like this we don't have a good signal when releasing the python packages.

**Comment by kpayson64** [comment]
@mehrdada will be looking into this one.

**Comment by jtattermusch** [comment]
@mehrdada   any progress here?

**Comment by mehrdada** [comment]
@jtattermusch not yet but I think I have some ideas

**Comment by mehrdada** [comment]
Looks like Docker images running the distribtests *never* had `python-dev` installed so naturally they cannot build from source!

Were these tests *ever* green on Linux? I'm be curious if they ever worked at all! :)

**Comment by mehrdada** [comment]
To clarify, it's not the case that the binary packages are failing to install. We explicitly test for source install **in addition** to binary wheel install in the distribtests.

**Comment by jtattermusch** [comment]
@mehrdada  those scripts definitely used to be green at some point (but that was because we only used to test the binary installation).

not having `python-dev` installed in the python test images is intentional to guarantee that we are testing installation from the binary wheel (and not compiling from source without knowing it).
Is there some other way to make sure that installation from binary packages is successful?

If we want to test both compilation from binary wheels and from source packages, we might need to split the distribtests into "binary" and "source" distribtests (each one using a different docker image)?

**Comment by mehrdada** [comment]
I agree that it should be separate, that is exactly what I plan to do now.

**Comment by jtattermusch** [comment]
Thanks for the fix!
